openpype/pipeline/publish/lib.py

```diff
@@ -690,3 +692,79 @@ def get_publish_repre_path(instance, repre, only_published=False):
     if os.path.exists(src_path):
         return src_path
     return None
+
+
+def get_custom_staging_dir_info(project_name, host_name, family, task_name,
+                                task_type, subset_name,
+                                project_settings=None,
+                                anatomy=None, log=None):
+    """Checks profiles if context should use special custom dir as staging.
+
+    Args:
+        project_name (str)
+        host_name (str)
+        family (str)
+        task_name (str)
+        task_type (str)
+        subset_name (str)
+        project_settings(Dict[str, Any]): Prepared project settings.
+        anatomy (Dict[str, Any])
+        log (Logger) (optional)
+
+    Returns:
+        (tuple)
+    Raises:
+        ValueError - if misconfigured template should be used
+    """
+    settings = project_settings or get_project_settings(project_name)
+    custom_staging_dir_profiles = (settings["global"]
+                                           ["tools"]
+                                           ["publish"]
+                                           ["custom_staging_dir_profiles"])
+    if not custom_staging_dir_profiles:
+        return None, None
+
+    if not log:
+        log = Logger.get_logger("get_custom_staging_dir_info")
+
+    filtering_criteria = {
+        "hosts": host_name,
+        "families": family,
+        "task_names": task_name,
+        "task_types": task_type,
+        "subsets": subset_name
+    }
+    profile = filter_profiles(custom_staging_dir_profiles,
+                              filtering_criteria,
+                              logger=log)
+
+    if not profile or not profile["active"]:
+        return None, None
```

Comment:
Shouldn't we be filtering the profiles on `active` state BEFORE filtering them since filtering profiles is based on finding the "best match". What if the first entry was a global matching profile (e.g. match anything) and it was set to inactive. Then it wouldn't ever match any of the profiles here because it could filter to that one and then ignore the others?

Same goes if e.g. I'd make a copy of one of my filter profiles that I like, but want to test a slightly different one - I'd deactivate the original, activate the new one but the profile filtering here might still be taking the inactive one.

Reply:
If the question is that  `filter_profiles` should filter (find best applicable) only in 'active' profiles is interesting question, but definitely for different PR/separate issue.

But I could totally see both use cases would be aplicable:
- I have general profile for `Nuke` - active
- I have more detailed profile for 'renders' in `Nuke`, I am publishing `render`, but custom dir doesnt' work, so I want to deactivate it for now to get `render` out, because chaos, deadline looming etc.etc.
- do I want to fall back to higher profile (I am still publishing from Nuke) and still run it or am I saying "for this context/profile doesn't use this" (because I think we are using latter use case in Settings to force some behavior on more descriptive profile - how would I really deactive it just for `render` family?)

Reply:
> If the question is that filter_profiles should filter (find best applicable) only in 'active' profiles is interesting question,

Not necessarily - it was in particular about this case but I didn't know there were more settings for profiles that had an "active" toggle. 

> 👉  **do I want to fall back to higher profile (I am still publishing from Nuke) and still run it** 👈  ~or am I saying "for this context/profile doesn't use this" (because I think we are using latter use case in Settings to force some behavior on more descriptive profile - how would I really deactive it just for render family?)~

That's what I would have expected to happen in this case based on the labels in settings, yes. I can understand your use case but that definitely isn't what I thought would happen.

It's more like an inclusive/exclusive filtering. Is active felt to me like it would just do nothing and not influence any of the other profiles defined.

Reply:
Well, I had it before as `use_transient_dir` and you didn't fancy that either, so I think we are running out of options here :D.

I created issue for that here https://github.com/ynput/OpenPype/issues/4681 , please feel free to comment there behavior you would expect.